### DISCUSSION TITLE
Writing assertions against original method vs. mock

### DISCUSSION THREAD

**Discussion by pawelgrzybek** [vitest-dev/vitest#5345] [General]
Hey.

This is purely just to find out what other folks do. In case that we have a method, and mock to this method in the same file, do you write your assertions against the original function/method name or against the mock? Example worth a 1000 words so [here is the StackBlitz example](https://stackblitz.com/edit/vitest-dev-vitest-5ts5q3?file=test%2Fsuite.test.ts) and the snippet below…

```ts
import { expect, test, vi } from 'vitest';
import { myMethod } from '../src/module.js';

const { mockedMyMethod } = vi.hoisted(() => ({ mockedMyMethod: vi.fn() }));

vi.mock('../src/module.ts', () => {
  return { myMethod: mockedMyMethod };
});

test('test example', () => {
  mockedMyMethod.mockReturnValue('i am mocked 🥑');

  expect(mockedMyMethod === myMethod).toBe(true);

  // Like this? 
  expect(mockedMyMethod()).toBe('i am mocked 🥑');

  // Or like that?
  expect(myMethod()).toBe('i am mocked 🥑');
});
```

Also, any advantage one over the other?

**Comment by hi-ogawa** [comment]
I don't know if I can say which is universally better in all situations, but for your simple example, I think you can skip `vi.hoisted` + `mockedMyMethod` entirely, so I guess the latter one looks more natural to me:
https://stackblitz.com/edit/vitest-dev-vitest-mje4rm?file=test%2Fsuite.test.ts

```ts
import { expect, test, vi } from 'vitest';
import { myMethod } from '../src/module.js';

vi.mock('../src/module.ts', () => {
  return { myMethod: vi.fn() };
});

test('test example', () => {
  // `vi.mocked` helper to get `vi.fn` equivalent typing
  vi.mocked(myMethod).mockReturnValue('i am mocked 🥑');

  expect(myMethod()).toBe('i am mocked 🥑');
});
```

**Reply by pawelgrzybek** [reply]
Oh yeah, I know about that. I tend to wrap all my module mocks in hoisted, purely for future-proofing. In case I need to change the implementation of a mock in another test.

Thanks for leaving your example here. It may be useful for future readers.
